### PR TITLE
[SQL] implement array_max, array_min functions

### PR DESCRIPTION
**Pull request by abhizer**
Is this a user-visible change (yes/no): yes

### CODE REVIEW

File: sql-to-dbsp-compiler/lib/sqllib/src/array.rs [hunk 1]
```diff
@@ -100,3 +100,59 @@ where
...
+{
+    vector.into_iter().flatten().min()
+}
+
+pub fn array_minNN<T>(vector: Option<Vec<Option<T>>>) -> Option<T>
+where
+    T: Ord,
+{
+    array_min_N(vector?)
+}
```
Comment (abhizer) [inline sql-to-dbsp-compiler/lib/sqllib/src/array.rs:158]: I thought about creating a macro, but maybe its better to do it the next time we have to create similar functions.

File: sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/ExpressionCompiler.java [hunk 2]
```diff
@@ -543,6 +543,21 @@ String getArrayCallName(RexCall call, DBSPExpression... ops) {
         return stringBuilder.toString();
     }
 
+    String getArrayCallNameWithElemNullability(RexCall call, DBSPExpression... ops) {
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/ExpressionCompiler.java:546]: please add a JavaDoc explanation, this function name is complicated.

File: sql-to-dbsp-compiler/lib/sqllib/src/array.rs [hunk 3]
```diff
@@ -100,3 +100,59 @@ where
...
+    T: Ord,
+{
+    vector.into_iter().min()
+}
+
+pub fn array_minN_<T>(vector: Option<Vec<T>>) -> Option<T>
+where
+    T: Ord,
+{
+    array_max__(vector?)
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/lib/sqllib/src/array.rs:143]: max?

Comment (mihaibudiu) [reply sql-to-dbsp-compiler/lib/sqllib/src/array.rs:143]: there are no tests for these functions, that's why this mistake went unnoticed.

**Review by mihaibudiu — CHANGES_REQUESTED** [review]

File: sql-to-dbsp-compiler/lib/sqllib/src/array.rs (hunk 3 above)
Comment (mihaibudiu) [reply sql-to-dbsp-compiler/lib/sqllib/src/array.rs:143]: there is also no documentation

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/ArrayTests.java [hunk 4]
```diff
@@ -415,4 +415,130 @@ public void testArrayAppendInnerNullable() {
...
+                new DBSPTupleExpression(
+                        new DBSPI32Literal(5, true)
+                )
+        );
+
+        this.testQuery(ddl, "SELECT ARRAY_MIN(val) FROM ARR_TBL",
+                new InputOutputChangeStream().addPair(new Change(input), new Change(result)));
+    }
+
+    @Test @Ignore("https://issues.apache.org/jira/browse/CALCITE-6275")
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/ArrayTests.java:513]: you should also link to an issue for our compiler

**Review by mihaibudiu — APPROVED** [review]

**Comment by mihaibudiu** [comment]
too many conflicts to fix in the web editor, please fix them and resubmit

**Pull request merged**